src/api/wallet/WalletApi.ts

```diff
@@ -43,6 +43,7 @@ export interface WalletInfo {
   isConnected: boolean
   userAddress?: string
   networkId?: number
+  blockNumber?: number
```

Comment:
I don't know if it's a good idea to add the block number here. This would update this object every time there's a new block number. 

Normally most components just need the userAddress or networkId, so they are reevaluated if these changes. Adding here the blockNumber might make this components either to redraw or to add some code to prevent it explicitly

Reply:
I believe that won't be a problem because if components don't need `blockNumber` they can just not subscribe to it and React is smart enough to not update those components. Not the case?

Reply:
But if it's there, then `addOnChangeWalletInfo` will register for any change in walletInfo, including the block

I feel it's strange that a block is part of the wallet info.

I would understand a separate event for the block